drivers/char/broadcom/vc_sm/vmcs_sm.c

```diff
@@ -1749,8 +1751,8 @@ static int vc_sm_ioctl_lock(struct SM_PRIV_DATA_T *private,
 			private->restart_sys = -EINTR;
 			private->int_action = VC_SM_MSG_TYPE_LOCK;
 			goto error;
-		} else if (status != 0 ||
-			   (status == 0 && result.res_mem == NULL)) {
+		} else if (status ||
+			   (!status && !(void *)result.res_mem)) {
```

Comment:
Isn't this cast pointless?

Reply:
Yes. Done.
Too many things passes through checkpatch.
res_mem being converted to u32 meant that comparison of u32 to NULL (ptr) became a warning. Then it prefers !(ptr) to ptr == NULL hence why we ended up here.

Just looking through and there are a bundle of ```if (X || (!X && Y)```, which either have redundant casts but can also be reduced to ```if (X || Y)```. I'll nuke those too.